src/ray/raylet/node_manager.cc

```diff
@@ -2418,26 +2550,39 @@ void NodeManager::AssignTask(const std::shared_ptr<Worker> &worker, const Task &
 bool NodeManager::FinishAssignedTask(Worker &worker) {
   TaskID task_id = worker.GetAssignedTaskId();
   RAY_LOG(DEBUG) << "Finished task " << task_id;
+  RAY_LOG(WARNING) << "FinishAssignedTask " << task_id;
 
-  // (See design_docs/task_states.rst for the state transition diagram.)
   Task task;
-  RAY_CHECK(local_queues_.RemoveTask(task_id, &task));
+  if (new_scheduler_enabled_) {
+    task = worker.GetAssignedTask();
+    leased_workers_.erase(worker.WorkerId()); // Maybe RAY_CHECK ???
```

Comment:
Wasn't this already erased in `HandleReturnWorker`? I don't think we want to erase it here.

Reply:
Correct. Deleted.